### PR TITLE
Submit coverage reports to Codecov

### PR DESCRIPTION
**Pull request by stevepeak**
Steve here from Codecov. Thank you for maintaining this project! I wanted to create this PR to add our coverage reporting tool to this repository, which we use at Codecov :+1: 

[View a report generated from this fork](https://codecov.io/github/stevepeak/python-oauth2?ref=604f8e0cb41c969a273aa85b02ea7857c5406e55)

I look forward to hearing your feedback. Let me know if you have any questions!

Cheers,
Steve

### CODE REVIEW

**Comment by rickhanlonii** [comment]
Hey Steve, 

Thanks for the PR! 

Does codecov post a comment on PRs as done above or can it hook into the [GitHub Status APIs](https://developer.github.com/v3/repos/statuses/)? Do you have badges we could add to the README as well?

**Comment by stevepeak** [comment]
Hey @rickhanlonii I've added the badge in the pull request. Github comments and statuses are enabled by default. You can change their settings in the Codecov UI by clicking the "gears" icon on https://codecov.io/gh/joestump/python-oauth2. You can disable the comment if you would like by toggling off the features [here](https://codecov.io/gh/joestump/python-oauth2/features/comment).

Thank you!

**Comment by jaitaiwan** [comment]
I'm pretty keen! @joestump I think this would need you to enable... unless you want to transfer this one into the Open Trust like previously discussed?

**Comment by stevepeak** [comment]
@jaitaiwan I can enable feature if you would like. Is there any questions you have about the service? Thank you!

**Comment by jaitaiwan** [comment]
I;m just waiting to make sure other contribs are happy for merge.

**Comment by rickhanlonii** [comment]
I'm +1 on the badge and -1 on the comment so I turned off the comment. Thanks @stevepeak.

File: .travis.yml [hunk 1]
```diff
@@ -4,5 +4,10 @@ python:
   - "2.7"
   - "3.3"
   - "3.4"
-install: "pip install -r requirements.txt"
-script: py.test 
+install:
+  - pip install -r requirements.txt
+  - pip install codecov pytest-cov
```
Comment (rickhanlonii) [inline .travis.yml]: @stevepeak, just curious: why not just add `codecov` and `pytest-cov` to requirements.txt?

**Comment by jaitaiwan** [comment]
@rickhanlonii could you put this into develop?

**Comment by rickhanlonii** [comment]
Was thinking the same, will do.

File: .travis.yml (hunk 1 above)
Comment (stevepeak) [inline .travis.yml]: I'm not sure how I missed that...Sorry. For sure move the into the requirements file.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+[![Join the chat at https://gitter.im/joestump/python-oauth2](https://img.shields.io/badge/gitter-join%20chat-1dce73.svg?style=flat-square)](https://gitter.im/joestump/python-oauth2?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) [![Build Status](http://img.shields.io/travis-ci/joestump/python-oauth2.png?branch=master&style=flat-square)](https://travis-ci.org/joestump/python-oauth2) [![Coverage](https://img.shields.io/codecov/c/github/joestump/python-oauth2.svg?style=flat-square)](https://codecov.io/gh/joestump/python-oauth2) ![Number of issues](https://img.shields.io/github/issues/joestump/python-oauth2.svg?style=flat-square) ![Licence MIT](https://img.shields.io/badge/license-MIT-blue.svg?style=flat-square)
```
Comment (1234-) [inline README.md]: # @codecov-io

**Pull request merged**